sentry-sidekiq/spec/sentry/sidekiq/sentry_context_middleware_spec.rb

```diff
@@ -0,0 +1,89 @@
+require "spec_helper"
+
+RSpec.shared_context "sidekiq", shared_context: :metadata do
+  let(:user) { { "id" => rand(10_000) } }
+
+  let(:client) do
+    Sidekiq::Client.new.tap do |client|
+      client.middleware do |chain|
+        chain.add Sentry::Sidekiq::SentryContextClientMiddleware
+      end
+    end
+  end
+
+  let(:random_empty_queue) do
+    Sidekiq::Queue.new(rand(10_000)).tap do |queue|
+      queue.clear
+    end
+  end
+end
+
+RSpec.describe Sentry::Sidekiq::SentryContextServerMiddleware do
+  include_context "sidekiq"
+
+  before do
+    perform_basic_setup do |config|
+      config.traces_sample_rate = 1.0
+    end
+  end
+
+  it "sets user to the current scope from the job" do
+    Sentry.set_user(user)
+
+    queue = random_empty_queue
+    options = { fetch: Sidekiq::BasicFetch.new(queues: [queue.name]) }
+    processor = Sidekiq::Processor.new(nil, options)
+
+    client.push('queue' => queue.name, 'class' => HappyWorker, 'args' => [])
+
+    expect { processor.send(:process_one) }.
+      to change { Sentry.get_current_client.transport.events.size }.by(1)
+
+    event =  Sentry.get_current_client.transport.events.first
+    expect(event).not_to be_nil
+    expect(event.user).to eq(user)
+  end
+
+  it "sets user to the current scope from the job even if worker raises an exception" do
+    Sentry.set_user(user)
+
+    queue = random_empty_queue
+    options = { fetch: Sidekiq::BasicFetch.new(queues: [queue.name]) }
+    processor = Sidekiq::Processor.new(nil, options)
+
+    client.push('queue' => queue.name, 'class' => SadWorker, 'args' => [])
+
+    expect do
+      begin; processor.send(:process_one); rescue RuntimeError; end
+    end.
+      to change { Sentry.get_current_client.transport.events.size }.by(1)
```

Comment:
This spec fails because two events are pushed. Honestly, I don't understand why. @st0012 Could you please help?

Reply:
if you enable tracing, it'll create a tracing event. and because you also use the `SadWorker`, which  raises an exception, it'll create another error event.

I think we should have 3 scenarios:

- Use `traces_sample_rate = 1` 
  - with `HappyWorker` to test the no exception scenario.
  - with `SadWorker` to test the exception with tracing scenario.
- Use `SadWorker` to test the exception scenario.

Reply:
Added those scenarios.